### PR TITLE
[CDP] Combined and OtherVA Alert messages

### PR DESCRIPTION
**Pull request by Dr-Pongo**
## Description
Updated medical-copays and debt-letters to include new alerts for Combined Portal update. These alerts reference the other debts/bills in the combined portal if there is a zero balance or error. There are currently calls inside of the applications to reference their sister apps, but these will be removed when the apps are migrated under the combined portal, and they will share the same store so the async checks will no longer be necessary. 

### Combined Alerts
department-of-veterans-affairs/va.gov-team#40885
### OtherVA Alerts
department-of-veterans-affairs/va.gov-team#40887

### debt-letters

---
<details><summary>VBA & VHA 404</summary>

image
</details>
<details><summary>VBA 404 & VHA balance</summary>

image
</details>
<details><summary>VBA 404 & VHA 0 balance</summary>

image
</details>
<details><summary>VBA 0 balance & VHA balance</summary>

image
</details>
<details><summary>VBA 0 balance & VHA 0 balance</summary>

image
</details>

---

### medical-copays
---
<details><summary>VHA & VBA 404</summary>

image
</details>
<details><summary>VHA 404 & VBA balance</summary>

image
</details>
<details><summary>VHA 404 & VBA 0 balance</summary>

image
</details>
<details><summary>VHA 0 balance & VBA balance</summary>

image
</details>
<details><summary>VHA 0 balance & VBA 0 balance</summary>

image
</details>

---

### Other VA info
<details><summary>medical-copays</summary>

image
</details>
<details><summary>debt-letters</summary>

image
</details>

## Acceptance criteria
- [x] Alerts populating to match design
   - [x] debt-letters
   - [x] medical-copays
- [x] Other VA showing alert message if 404
   - [x]  debt-letters
   - [x] medical-copays

### CODE REVIEW

File: src/applications/debt-letters/components/DebtLettersSummary.jsx [hunk 1]
```diff
@@ -53,9 +62,45 @@ const EmptyItemsAlert = () => (
...
+        <>
+          <h3 className="vads-u-font-size--h4">{alertInfo.secondHeader}</h3>
+          {alertInfo.secondBody}
+        </>
+      )}
+    </va-alert>
+  );
+};
+
+const fetchCopaysResponseAsync = async () => {
```
Comment (mdewey) [inline src/applications/debt-letters/components/DebtLettersSummary.jsx:85]: ORGANIZATION: Eventually these stray API calls should in one place.

File: src/applications/medical-copays/components/AlertView.jsx [hunk 2]
```diff
@@ -2,8 +2,39 @@ import React from 'react';
...
+import alertMessage from '../../combined-debt-portal/combined/utils/alert-messages';
+import OtherVADebts from './OtherVADebts';
+import {
+  ALERT_TYPES,
+  APP_TYPES,
+  API_RESPONSES,
+} from '../../combined-debt-portal/combined/utils/helpers';
 
-const AlertView = ({ pathname, alertType, error }) => {
+const renderAlert = (alertType, hasDebts) => {
```
Comment (mdewey) [inline src/applications/medical-copays/components/AlertView.jsx:13]: NITPICK: I see this looks very similar to the renderAleryin the other file. Might be a point to refactor later

File: src/applications/medical-copays/containers/OverviewPage.jsx [hunk 3]
```diff
@@ -39,6 +24,30 @@ const OverviewPage = () => {
   const statementsByUniqueFacility = uniqBy(sortedStatements, 'pSFacilityNum');
   const title = 'Current copay balances';
 
+  const renderOtherVA = () => {
+    const alertInfo = alertMessage(ALERT_TYPES.ERROR, APP_TYPES.DEBT);
+    if (hasDebts > 0) {
+      return <OtherVADebts module={APP_TYPES.DEBT} />;
+    }
+    if (hasDebts === API_RESPONSES.ERROR) {
```
Comment (mdewey) [inline src/applications/medical-copays/containers/OverviewPage.jsx:32]: NITPICK: Same as above with the guard clause

**Review by mdewey — APPROVED** [review]
Pretty good, some minor feedback. Feel free to clean some of the things up, but its good to go

**Review by OptionSelect — APPROVED** [review]
LGTM! 🚀

**Pull request merged**